valhalla/proposals/accounting.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import requests
+from django.conf import settings
+
+from valhalla.proposals.models import Semester
+
+logger = logging.getLogger(__name__)
+
+
+def split_time(start, end, chunks=4):
+    spans = []
+    chunk = (end - start) / chunks
+    spans.append((start, start + chunk))
+    for i in range(0, chunks - 1):
+        spans.append((spans[i][1], spans[i][1] + chunk))
+    return spans
+
+
+def get_time_totals_from_pond(timeallocation, start, end, too, recur=0):
+    """
+    Pond queries are too slow with large proposals and time out, this hack splits the query
+    when a timeout occurs
+    """
+    if recur > 3:
+        raise RecursionError('Pond is timing out, too much recursion.')
+
+    total = 0
+    try:
+        total += query_pond(timeallocation.proposal.id, start, end, timeallocation.telescope_class, too)
+    except requests.HTTPError:
+        logger.warning('We got a pond inception. Splitting further.')
```

Comment:
Output from a real run: 

Attempting to get time used for auto_focus from 2016-10-01 00:00:00+00:00 to 2017-03-31 23:59:59+00:00

We got a pond inception. Splitting further.

Attempting to get time used for auto_focus from 2016-10-01 00:00:00+00:00 to 2016-11-15 11:59:59.750000+00:00

Attempting to get time used for auto_focus from 2016-11-15 11:59:59.750000+00:00 to 2016-12-30 23:59:59.500000+00:00

Attempting to get time used for auto_focus from 2016-12-30 23:59:59.500000+00:00 to 2017-02-14 11:59:59.250000+00:00

Attempting to get time used for auto_focus from 2017-02-14 11:59:59.250000+00:00 to 2017-03-31 23:59:59+00:00

Attempting to get time used for auto_focus from 2016-10-01 00:00:00+00:00 to 2017-03-31 23:59:59+00:00